mesonbuild/modules/_qt.py

```diff
@@ -631,3 +743,428 @@ def compile_translations(self, state: 'ModuleState', args: T.Tuple, kwargs: 'Com
+            }
+            collected_json = self._moc_json_collect(state, moc_collect_json_kwargs)
+            output.append(collected_json)
+
+        typeinfo_file: str = ''
+        #cmake NO_GENERATE_QMLTYPE disable the whole type registration, not just the .qmltype generation
+        if kwargs['generate_qmltype']:
+            qmltyperegistrar_kwargs: GenQmlTypeRegistrarKwArgs = {
+                'target_name': target_name,
+                'import_name': module_name,
+                'major_version': module_version_major,
+                'minor_version': module_version_minor,
+                'collected_json': collected_json,
+                'namespace': kwargs['namespace'],
+                'generate_qmltype': True,
+                'extra_args': kwargs['qmltyperegistrar_extra_args'],
+                'typeinfo': kwargs['typeinfo'],
+                'method': kwargs['method'],
+                'install': kwargs['install'],
+                'install_dir': module_install_dir,
+            }
+            type_registrar_output = self._qml_type_registrar(state, qmltyperegistrar_kwargs)
+            output.append(type_registrar_output)
+            if len(type_registrar_output.get_outputs()) == 2:
+                typeinfo_file = type_registrar_output.get_outputs()[1]
+
+        if kwargs['generate_qmldir']:
+            qmldir_kwargs: GenQmldirKwArgs = {
+                'output': f'{target_name}_qmldir',
+                'module_name': module_name,
+                'module_version': module_version_short,
+                'qml_sources': kwargs['qml_sources'],
+                'qml_singletons': kwargs['qml_singletons'],
+                'qml_internals': kwargs['qml_internals'],
+                'imports': kwargs['imports'],
+                'optional_imports': kwargs['optional_imports'],
+                'default_imports': kwargs['default_imports'],
+                'depends_imports': kwargs['depends_imports'],
+                'designer_supported': kwargs['designer_supported'],
+                'typeinfo': typeinfo_file,
+                'module_prefix': module_prefix_full,
+            }
+            qmldir_file: File = self._gen_qmldir(state, qmldir_kwargs)
+
+            qmldir_qrc_kwargs: GenQrcKwArgs = {
+                'output': f'{target_name}_qmldir.qrc',
+                'sources': self._source_to_files(state, [qmldir_file]),
+                'aliases': ['qmldir'],
+                'prefix': module_prefix_full,
+            }
+            qrc_resouces.append(self._gen_qrc(state, qmldir_qrc_kwargs))
+
+            if kwargs['install']:
+                self.interpreter.install_data_impl([qmldir_file], module_install_dir,
+                                                   FileMode(), ['qmldir'], 'devel')
+
+        if qrc_resouces:
+            compile_resource_kwargs: ResourceCompilerKwArgs = {
+                'name': target_name,
+                'sources': qrc_resouces,
+                'extra_args': [],
```

Comment:
I seem to have a case, when I want to pass `--no-zstd` here. Would it be too hard to add this argument to the qml_module() call?